src/test/java/org/broadinstitute/consent/http/service/ElasticSearchServiceTest.java

```diff
@@ -632,16 +647,46 @@ void testIndexStudyWithDatasets() {
     when(datasetDAO.findDatasetsByIdList(any())).thenReturn(List.of(d));
 
     initService();
-    assertDoesNotThrow(() -> service.indexStudy(1));
+    assertDoesNotThrow(() -> service.indexStudy(1, user));
   }
 
   @Test
   void testIndexStudyWithNoDatasets() {
+    User user = new User();
+    user.setUserId(1);
     Study study = new Study();
     study.setStudyId(1);
     when(studyDAO.findStudyById(any())).thenReturn(study);
 
     initService();
-    assertDoesNotThrow(() -> service.indexStudy(1));
+    assertDoesNotThrow(() -> {
+      try (var response = service.indexStudy(1, user)) {
+        assertEquals(HttpStatusCodes.STATUS_CODE_NOT_FOUND, response.getStatus());
+      }
+    });
   }
+
+  @Test
+  void testUpdateDatasetIndexWithValue() {
+    initService();
+    assertDoesNotThrow(() -> service.updateDatasetIndex(1, 1, Instant.now()));
+  }
+
+  @Test
+  void testDeleteDatasetIndexWhenDatasetExists() throws Exception {
+    Dataset dataset = new Dataset();
+    when(datasetDAO.findDatasetById(any())).thenReturn(dataset);
+    initService();
+    assertDoesNotThrow(() -> service.updateDatasetIndex(1, 1, null));
+    verify(datasetServiceDAO, times(1)).updateDatasetIndex(any(), any(), any());
```

Comment:
`times(1)` is redundant here
```suggestion
    verify(datasetServiceDAO).updateDatasetIndex(any(), any(), any());
```